### PR TITLE
fix: Covert hex currency codes (#88)

### PR DESCRIPTION
**Pull request by ckniffen**
If currency code is 40 characters convert it to ASCII

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

### Before
Screen Shot 2022-04-14 at 1 53 57 PM

Screen Shot 2022-04-14 at 1 54 03 PM

### After
Screen Shot 2022-04-14 at 12 54 45 PM

Screen Shot 2022-04-14 at 12 54 29 PM

## Test Plan

This will not impact currency pair output or any place currently not using the Currency component.

### CODE REVIEW

**Review by mvadari — APPROVED** [review]
Can you add screenshots to the PR of the before/after with an example of this code change?

**Comment by ckniffen** [comment]
> Can you add screenshots to the PR of the before/after with an example of this code change?

Done

File: src/containers/shared/components/test/Currency.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  it('handles currency codes that are 4 characters ', () => {
+    const wrapper = mount(
+      <Currency currency={"WOOT"} amount={5.001} />
+    );
+    expect(wrapper.find('.currency').text()).toEqual('WOOT');
+    wrapper.unmount();
+  });
+
+  it('handles currency codes that are 4 characters ', () => {
```
Comment (khancode) [inline src/containers/shared/components/test/Currency.test.js:23]:
```suggestion
  it('handles currency codes that are 40 characters ', () => {
```

File: src/containers/shared/components/test/Currency.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const wrapper = mount(
+      <Currency currency="584D455441000000000000000000000000000000" amount={5.001} />
+    );
+    expect(wrapper.find('.currency').text()).toEqual('XMETA');
+    wrapper.unmount();
+  });
+
+  it('handles currency codes that are 4 characters and and issuer ', () => {
```
Comment (khancode) [inline src/containers/shared/components/test/Currency.test.js:31]:
```suggestion
  it('handles currency codes that are 40 characters with issuer ', () => {
```

File: src/containers/shared/components/Currency.tsx [hunk 3]
```diff
@@ -10,14 +13,17 @@ interface Props {
...
+  const currencyCode =
+    currency?.length === NON_STANDARD_CODE_LENGTH ? hexToString(currency) : currency;
+
   const content = issuer ? (
     <>
-      {currency}
+      {currencyCode}
       .
       <Account account={issuer} link={link} />
     </>
```
Comment (khancode) [inline src/containers/shared/components/Currency.tsx:24]: I'm surprised you can use empty tags `</>` in React. What do they get rendered as by default? Just curious 😀

**Review by JST5000 — APPROVED** [review]
Looks good to me! (You'll see me type the acronym LGTM in future PR's a lot :P)

**Review by khancode — APPROVED** [review]
LGTM once you address the small typos.

File: src/containers/shared/components/Currency.tsx (hunk 3 above)
Comment (ckniffen) [reply src/containers/shared/components/Currency.tsx:24]: They look to be rendered as nothing.  Seems to be a way to get around having a wrapping element.

**Pull request merged**